Library/Homebrew/cmd/uninstall.rb

```diff
@@ -59,6 +73,48 @@ def uninstall
     end
   end
 
+  def check_for_dependents(kegs)
+    return false unless result = find_some_installed_dependents(kegs)
+
+    requireds, dependents = result
+
+    msg = "Refusing to uninstall #{requireds.join(", ")} because "
+    msg << (requireds.count == 1 ? "it is" : "they are")
+    msg << " required by #{dependents.join(", ")}, which "
+    msg << (dependents.count == 1 ? "is" : "are")
+    msg << " currently installed."
+    ofail msg
+    print "You can override this and force removal with "
+    puts "`brew uninstall --ignore-dependencies #{requireds.map(&:name).join(" ")}`."
+
+    true
+  end
+
+  # Will return some kegs, and some dependencies, if they're present.
+  # For efficiency, we don't bother trying to get complete data.
+  def find_some_installed_dependents(kegs)
+    kegs.each do |keg|
+      dependents = keg.installed_dependents - kegs
+      dependents.map! { |d| "#{d.name} #{d.version}" }
+      return [keg], dependents if dependents.any?
+    end
+
+    # Find formulae that didn't have dependencies saved in all of their kegs,
+    # so need them to be calculated now.
+    #
+    # This happens after the initial dependency check because it's sloooow.
+    remaining_formulae = Formula.installed.select { |f|
+      f.installed_kegs.any? { |k| Tab.for_keg(k).runtime_dependencies.nil? }
+    }
+    Diagnostic.missing_deps(remaining_formulae, kegs.map(&:name)) do |dependent, required|
```

Comment:
Probably want to extract this method from `Diagnostic` into somewhere else that makes sense to be shared.